pkg/agent/route/route_windows.go

```diff
@@ -141,8 +144,19 @@ func (c *Client) AddRoutes(podCIDR *net.IPNet, nodeName string, peerNodeIP, peer
 		return nil
 	}
 
-	if err := c.nr.NewNetRoute(route.LinkIndex, route.DestinationSubnet, route.GatewayAddress); err != nil {
-		return err
+	if isSvc {
+		if err := util.NewNetRouteWithMetric(c.nodeConfig.GatewayConfig.LinkIndex, podCIDR, peerGwIP, metricNormal); err != nil &&
+			!strings.Contains(err.Error(), "The object already exists") {
+			return err
+		}
+		// Remove the Service route on host by kube-proxy
+		if err := util.RemoveInterfaceAddress("vEthernet (HNS Internal NIC)", podCIDR.IP); err != nil {
+			return err
+		}
```

Comment:
kube-proxy doesn't try to re-assign the IP after we remove it?

Reply:
I observed that kube-proxy doesn't assign again.

"TODO" added. Route with lower metric will be used. I observed that just adding a route with lower metric is not enough. `Remove-NetIPAddress` for the one by kube-proxy is also needed.